mcs/class/System/System.Net/MacProxy.cs

```diff
@@ -138,16 +177,19 @@ static unsafe CFArray Create (params IntPtr[] values)
 				return new CFArray (handle, false);
 			}
 		}
+		
+		public static CFArray CreateArray (params INativeObject[] values)
+		{
+			return new CFArray (Create (values), false);
```

Comment:
It looks like `Create` returns a retained instance (it's what `CFArrayCreate` does at least), which means you should pass `owns = true` here.